deployer/src/lib.rs

```diff
@@ -91,6 +94,13 @@ pub struct DeployerService<D: Dal + Send + Sync + 'static> {
     deployment_state_machine_channel:
         tokio::sync::mpsc::Sender<Box<dyn Task<(), Output = (), Error = project::error::Error>>>,
     runtime_start_channel: Sender<RunnableDeployment>,
+
+    // These channel are intended to work with one subscriber. They can be reinitialized as well.
+    // They are wrapped in Mutex first because we want them with interior muttability (see unsubscribe method).
+    // They are wrapped in Arc because they get sent on all deployment tasks.
+    project_events_channel_tx: Arc<Mutex<Option<UnboundedSender<ProjectEvent>>>>,
+    project_events_channel_rx: Arc<Mutex<Option<UnboundedReceiver<ProjectEvent>>>>,
```

Comment:
Why do we only want one subscriber?

Reply:
The idea is that only the gateway subscribes at this moment to the deployer project events. We would want eventually to support more than one subscriber, but if we want it we should keep track of each of them, and handle cleaning up connections if they get an EOF as well.

Reply:
Maybe we can put only the `Receiver` behind a mutex then?

Reply:
Nvm, you will need the mutex to reset the channel